core-bundle/tests/Image/Studio/FigureBuilderTest.php

```diff
@@ -0,0 +1,1007 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Image\Studio;
+
+use Contao\CoreBundle\Exception\InvalidResourceException;
+use Contao\CoreBundle\File\MetaData;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Image\Studio\Figure;
+use Contao\CoreBundle\Image\Studio\FigureBuilder;
+use Contao\CoreBundle\Image\Studio\ImageResult;
+use Contao\CoreBundle\Image\Studio\LightBoxResult;
+use Contao\CoreBundle\Image\Studio\Studio;
+use Contao\CoreBundle\Tests\TestCase;
+use Contao\FilesModel;
+use Contao\Image\ImageInterface;
+use Contao\PageModel;
+use Contao\System;
+use Contao\Validator;
+use PHPUnit\Framework\MockObject\MockObject;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Webmozart\PathUtil\Path;
+
+class FigureBuilderTest extends TestCase
+{
+    public function testFromFilesModel(): void
+    {
+        [$absoluteFilePath, $relativeFilePath] = $this->getTestFilePaths();
+
+        /** @var FileSModel&MockObject $model */
+        $model = $this->mockClassWithProperties(FilesModel::class);
+        $model->type = 'file';
+        $model->path = $relativeFilePath;
+
+        $studio = $this->getStudioMockForImage($absoluteFilePath);
+
+        $this->getFigureBuilder($studio, null)
+            ->fromFilesModel($model)
+            ->build()
+        ;
+    }
+
+    public function testFromFilesModelFailsWithInvalidDBAFSType(): void
+    {
+        /** @var FileSModel&MockObject $model */
+        $model = $this->mockClassWithProperties(FilesModel::class);
+        $model->type = 'folder';
+
+        $this->expectException(InvalidResourceException::class);
```

Comment:
I think this is too early here, because it will also catch exceptions in the `getFigureBuilder()` method. Or is this intended?

Reply:
Adressed in 364bd38e280f2239aa1769f39aab0dfb25f9204a.